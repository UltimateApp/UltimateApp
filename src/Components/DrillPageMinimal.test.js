import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';
import { connect, Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import fixtures from '../Fixtures/TestFixtures';
import store from '../Store/testStore';

import ConnectedDrillPageMinimal, { DrillPageMinimal } from './DrillPageMinimal';

afterEach(cleanup);
beforeEach(() => jest.useFakeTimers()); // for Animated
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<DrillPageMinimal />', () => {
  const navigation = { navigate: jest.fn(), setOptions: jest.fn() };
  const program = fixtures.programs[0];
  const training = program.trainings[0];
  const drill = training.drills[0];
  let route = {
    params: {
      program,
      training,
      drill,
    },
  };

  it('renders correctly with a training and a drill', () => {
    const tree = renderer.create(<DrillPageMinimal navigation={navigation} route={route} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('links to next drill within training when finished', async () => {
    const { getByText } = render(<DrillPageMinimal navigation={navigation} route={route} />);

    await fireEvent.press(getByText('Next drill'));

    const secondDrill = training.drills[1];
    expect(navigation.navigate).toBeCalledWith('DrillPageMinimal', { drill: secondDrill, training, program });
  });

  it('links to programs list page when finished and is last drill', async () => {
    const completeTraining = jest.fn();
    route = {
      params: {
        program,
        training,
        drill: training.drills[training.drills.length - 1],
      },
    };

    const { getByText } = render(
      <DrillPageMinimal navigation={navigation} route={route} completeTraining={completeTraining} />,
    );

    await fireEvent.press(getByText('Finish Training!'));

    expect(completeTraining).toBeCalledWith({ training, program });
    expect(navigation.navigate).toBeCalledWith('ProgramListPage', { activeProgram: program.id });
  });
});
