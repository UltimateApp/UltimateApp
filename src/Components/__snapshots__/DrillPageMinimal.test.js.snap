// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillPageMinimal /> renders correctly with a training and a drill 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Drill 
        1
         - 
        Fitness Drill Title
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(150, 150, 150)",
                "height": 5,
                "left": 0,
                "position": "absolute",
                "top": 5,
                "width": "100%",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "rgb(0, 0, 0)",
                "height": 5,
                "left": 0,
                "position": "absolute",
                "top": 5,
                "width": "0%",
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgb(0, 0, 0)",
                "borderRadius": 10,
                "borderWidth": 0,
                "height": 15,
                "opacity": 1,
                "width": 15,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgb(150, 150, 150)",
                "borderRadius": 10,
                "borderWidth": 0,
                "height": 15,
                "opacity": 1,
                "width": 15,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "maxHeight": "20%",
          "padding": 5,
          "paddingLeft": 20,
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#232b2b",
              "flexGrow": 0,
              "flexShrink": 0,
              "fontWeight": "bold",
              "width": "25%",
            }
          }
        >
          Goals
        </Text>
        <Text
          style={
            Object {
              "color": "#A6A6A6",
              "flexGrow": 1,
            }
          }
        >
          Legs
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "maxHeight": "20%",
          "padding": 5,
          "paddingLeft": 20,
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#232b2b",
              "flexGrow": 0,
              "flexShrink": 0,
              "fontWeight": "bold",
              "width": "25%",
            }
          }
        >
          Equipment
        </Text>
        <Text
          style={
            Object {
              "color": "#A6A6A6",
              "flexGrow": 1,
            }
          }
        >
          Equipment needs for the drill
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "maxHeight": "20%",
          "padding": 5,
          "paddingLeft": 20,
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#232b2b",
              "flexGrow": 0,
              "flexShrink": 0,
              "fontWeight": "bold",
              "width": "25%",
            }
          }
        >
          Description
        </Text>
        <Text
          style={
            Object {
              "color": "#A6A6A6",
              "flexGrow": 1,
            }
          }
        >
          Description of the drill
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": 1334,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <RNCWebView
            cacheEnabled={true}
            javaScriptEnabled={true}
            messagingEnabled={false}
            onContentProcessDidTerminate={[Function]}
            onHttpError={[Function]}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingProgress={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={[Function]}
            source={
              Object {
                "uri": "https://www.youtube.com/embed/JkVHrA5o23o",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
              ]
            }
            useSharedProcessPool={true}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#232b2b",
            "opacity": 1,
            "padding": 20,
            "textAlign": "center",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(255, 255, 255)",
            }
          }
        >
          More details on this drill
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
